projects/smart-table/src/lib/components/column-selector/column-selector.component.ts

```diff
@@ -0,0 +1,87 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {ConfigurationService} from '../../services/configuration.service';
+import {combineLatest, merge, Observable, Subject} from 'rxjs';
+import {SmartTableColumnConfig, SmartTableConfig} from '../../smart-table.types';
+import {first, map, shareReplay, switchMap, tap} from 'rxjs/operators';
+import {FlyoutService} from '@acpaas-ui/ngx-flyout';
+import {CdkDragDrop, moveItemInArray} from '@angular/cdk/drag-drop';
+import {sortColumn} from '../../helper/helpers';
+
+@Component({
+  selector: 'aui-table-column-selector',
+  templateUrl: './column-selector.component.html',
+  styleUrls: ['./column-selector.component.scss']
+})
+export class TableColumnSelectorComponent implements OnInit {
+  @Input()
+  instanceId: string;
+  configuration$: Observable<SmartTableConfig>;
+
+  pendingColumnOperation$: Observable<SmartTableColumnConfig[]>;
+  toggleColumnsVisibility$ = new Subject<string>();
+  updateSortIndexByKey$ = new Subject<{ oldIndex: number, newIndex: number }>();
+
+  constructor(
+    private configurationService: ConfigurationService,
+    private flyoutService: FlyoutService) {
+  }
+
+  ngOnInit() {
+    this.configuration$ = this.configurationService.getConfiguration(this.instanceId);
+    this.pendingColumnOperation$ = merge(
+      this.configuration$.pipe(map(config => [...config.columns.sort(sortColumn)])),
+      this.toggleColumnsVisibility$.pipe(
+        switchMap((key: string) => this.pendingColumnOperation$.pipe(
+          first(),
+          map(columns => {
+            const newColumns = [...columns];
```

Comment:
zo worden de column objecten zelf wel niet gecloned. Als je hieronder de visible prop aanpast zal dit ook effect hebben op de columns.

Reply:
Dat is ok, het is alleen de bedoeling om de ref te veranderen van de array